htmd/qm/gaussian.py

```diff
@@ -0,0 +1,171 @@
+
+    def _completed(self, directory):
+        return os.path.exists(os.path.join(directory, 'g09.out'))
+
+    def _writeInput(self, directory, iframe):
+
+        with open(os.path.join(directory, 'g09.in'), 'w') as f:
+
+            f.write('%nprocshared = {}\n'.format(self._ncpus))
+            f.write('%mem = {} GB\n\n'.format(self._mem))
+
+            if self.theory == 'HF':
+                dispersion = ''
+            elif self.theory == 'B3LYP':
+                dispersion = ' EmpiricalDispersion=GD3'
+            else:
+                raise NotImplementedError
+
+            line = '# %s/%s NoSymm SCF=tight' % (self.theory, self.basis)
+            line += dispersion
+
+            if self.solvent == 'vacuum':
+                pass
+            elif self.solvent == 'PCM':
+                line += ' SCRF=PCM'
+            else:
+                raise NotImplementedError
+
+            if self.optimize:
+                line += ' Opt=ModRedundant'
+
+            if self.esp_points is not None:
+                line +=  ' Prop=(read,field)'
+
+            line += '\n\n'
+            f.write(line)
+
+            f.write('Mol\n%d %d\n' % (self._charge, self.multiplicity))
+            elements = self._molecule.element
+            coords = self._molecule.coords[:, :, iframe]
+            for element, coord in zip(elements, coords):
+                f.write('    %-2s %10f %10f %10f\n' % (element, coord[0], coord[1], coord[2]))
+            f.write('\n')
+
+            if self._restrained_dihedrals is not None:
+                for dihedral in self._restrained_dihedrals:
+                    f.write('%d %d %d %d F\n' % tuple(dihedral))
+                f.write('\n')
+
+            if self.esp_points is not None:
+                f.write('@grid.dat /N\n')
+
+    def _readOutput(self, directory):
+
+        result = QMResult()
+        result.completed = True
+
+        with open(os.path.join(directory, 'g09.out')) as f:
+            fl = f.readlines()
+
+        try:
+            data = {}
+
+            for i in range(len(fl)):
+                if "Dipole moment (field-independent basis, Debye):" in fl[i]:
+                    s = fl[i + 1].split()
+                    data['dipole'] = [float(s[1]), float(s[3]), float(s[5]), float(s[7])]
+                if "Traceless Quadrupole moment (field-independent basis, Debye-Ang):" in fl[i]:
+                    s1 = fl[i + 1].split()
+                    s2 = fl[i + 2].split()
+                    data['quadrupole'] = [float(s1[1]), float(s1[3]), float(s1[5]), float(s2[1]), float(s2[3]),
+                                          float(s2[5])]
+                if "Mulliken atomic charges:" in fl[i] or "Mulliken charges:" in fl[i]:
+                    data['mulliken'] = []
+                    for j in range(self._natoms):
+                        data['mulliken'].append(float(fl[i + 2 + j].split()[2]))
+
+            for l in fl:
+                if "SCF Done:  E(RHF) = " in l:
```

Comment:
Maybe both at the same time with regex? Or maybe not worth it.

Reply:
I cannot test with Gaussian, so no point to change.